include/vkd3d_swapchain_factory.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Joshua Ashton for Valve Software
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ */
+
+#ifndef __VKD3D_SWAPCHAIN_FACTORY_H
+#define __VKD3D_SWAPCHAIN_FACTORY_H
+
+typedef interface IDXGIFactory IDXGIFactory;
+typedef struct DXGI_SWAP_CHAIN_DESC1 DXGI_SWAP_CHAIN_DESC1;
+typedef struct DXGI_SWAP_CHAIN_FULLSCREEN_DESC DXGI_SWAP_CHAIN_FULLSCREEN_DESC;
+typedef interface IDXGIOutput IDXGIOutput;
+typedef interface IDXGISwapChain1 IDXGISwapChain1;
+
+typedef interface IWineDXGISwapChainFactory IWineDXGISwapChainFactory;
+
+DEFINE_GUID(IID_IWineDXGISwapChainFactory, 0x53cb4ff0,0xc25a,0x4164,0xa8,0x91,0x0e,0x83,0xdb,0x0a,0x7a,0xac);
```

Comment:
Wouldn't this normally be part of an IDL header?

Reply:
It doesn't really work out because our DXGI stuff doesn't match up.

Reply:
I'll try it

Reply:
Resolved locally.